src/backend/vault-hsm/dh-curve448.c

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2016 Southern Storm Software, Pty Ltd.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+#include "internal.h"
+#include "crypto/curve448/curve448.h"
+#include <string.h>
+
+typedef struct
+{
+    struct NoiseDHState_s parent;
+    uint8_t private_key[56];
+    uint8_t public_key[56];
+
+} NoiseCurve448State;
+
+/* Curve448 base point from RFC 7748, 5 in little-endian order */
+static uint8_t const basepoint[56] = {5};
+
+static int noise_curve448_generate_keypair
+    (NoiseDHState *state, const NoiseDHState *other)
```

Comment:
Since `other` is not used, shan't we check that it is set to NULL and return an error if it's not?